src/adapters/fauna/index.js

```diff
@@ -0,0 +1,420 @@
+      )
+    }
+
+    async function createSession (user) {
+      console.log('-----------createSession------------')
+      // console.log(user);
+      let expires = null
+      if (sessionMaxAge) {
+        const dateExpires = new Date()
+        dateExpires.setTime(dateExpires.getTime() + sessionMaxAge)
+        expires = dateExpires.toISOString()
+      }
+
+      return faunaWrapper(
+        q.Select(
+          'data',
+          q.Create(q.Collection('sessions'), {
+            data: {
+              expires,
+              userId: user.id,
+              sessionToken: randomBytes(32).toString('hex'),
+              accessToken: randomBytes(32).toString('hex'),
+              id: uuidv4(),
+              createdAt: Date.now(),
+              updatedAt: Date.now()
+            }
+          })
+        ),
+        'CREATE_SESSION_ERROR'
+      )
+    }
+
+    async function getSession (sessionToken) {
+      console.log('-----------getSession------------')
+      // console.log(sessionToken);
+
+      const session = await serverClient.query(
+        q.Let(
+          {
+            ref: q.Match(q.Index(INDEX_SESSIONS_SESSION_TOKEN), sessionToken)
+          },
+          q.If(q.Exists(q.Var('ref')), q.Select('data', q.Get(q.Var('ref'))), null)
+        )
+      )
+      // Check session has not expired (do not return it if it has)
+      if (session && session.expires && new Date() > session.expires) {
+        await serverClient.query(
+          q.Delete(
+            q.Select(
+              'ref',
+              q.Get(q.Match(q.Index(INDEX_SESSIONS_SESSION_TOKEN), sessionToken))
+            )
+          )
+        )
+        return null
+      }
+
+      return session
+    }
+
+    async function updateSession (session, force) {
+      console.log('-----------updateSession------------')
+      // console.log(session, force);
+      if (sessionMaxAge && (sessionUpdateAge || sessionUpdateAge === 0) && session.expires) {
+        // Calculate last updated date, to throttle write updates to database
+        // Formula: ({expiry date} - sessionMaxAge) + sessionUpdateAge
+        //     e.g. ({expiry date} - 30 days) + 1 hour
+        //
+        // Default for sessionMaxAge is 30 days.
+        // Default for sessionUpdateAge is 1 hour.
+        const dateSessionIsDueToBeUpdated = new Date(session.expires)
+        dateSessionIsDueToBeUpdated.setTime(
+          dateSessionIsDueToBeUpdated.getTime() - sessionMaxAge
+        )
+        dateSessionIsDueToBeUpdated.setTime(
+          dateSessionIsDueToBeUpdated.getTime() + sessionUpdateAge
+        )
+
+        // Trigger update of session expiry date and write to database, only
+        // if the session was last updated more than {sessionUpdateAge} ago
+        if (new Date() > dateSessionIsDueToBeUpdated) {
+          const newExpiryDate = new Date()
+          newExpiryDate.setTime(newExpiryDate.getTime() + sessionMaxAge)
+          session.expires = newExpiryDate
```

Comment:
Shouldn't this be?
``session.expires = newExpiryDate.valueOf()``
If not fauna just stores an empty object (at least that's what I've been getting)

Reply:
The logic is kept in sync with other adapters: https://github.com/nextauthjs/next-auth/tree/main/src/adapters
Let me check why fauna isn't saving this